File: beginner/javascript/2139.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.on('line', (input) => {
  const [mes, dia] = input.split(' ').map(Number);
  const diasNoMes = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 25];
  let diasFaltando;

  if (mes === 12 && dia === 25) {
    console.log('E natal!');
  } else if (mes === 12 && dia === 24) {
    console.log('E vespera de natal!');
  } else if (mes === 12 && dia > 25) {
    console.log('Ja passou!');
  } else {
    diasFaltando = diasNoMes[mes - 1] - dia;
    for (let i = mes; i < 12; i++) {
      diasFaltando += diasNoMes[i];
    }
    console.log(`Faltam ${diasFaltando} dias para o natal!`);
  }
});

File: beginner/python/2139.py
while True:
    try: 
        mes, dia = list(map(int, input().split()))
        
        dias_no_mes = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 25]

        if (mes == 12 and dia == 25):
            print("E natal!")
        elif (mes == 12 and dia == 24): 
            print("E vespera de natal!")
        elif (mes == 12 and dia > 25):
            print("Ja passou!")
        else:
            dias_faltando = dias_no_mes[mes - 1] - dia
            for i in range(mes,12): 
                dias_faltando += dias_no_mes[i]
           
            print("Faltam %d dias para o natal!" % dias_faltando)
    except EOFError:
        break
   